lib/SILGen/SILGenTopLevel.h

```diff
@@ -0,0 +1,75 @@
+//===--- SILGenToplevel.h - Function Specific AST lower context -*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_SILGEN_SILGENTOPLEVEL_H
+#define SWIFT_SILGEN_SILGENTOPLEVEL_H
+
+#include "SILGen.h"
+#include "swift/AST/ASTVisitor.h"
+#include "swift/AST/SourceFile.h"
+#include "swift/AST/TypeMemberVisitor.h"
+
+namespace swift {
+
+namespace Lowering {
+
+/// Generates a `SILFunction` for toplevel code
```

Comment:
A bit more detail would be nice; script mode, TopLevelCodeDecl, global variable initializers, etc

Reply:
Makes sense, thanks! Does it look better now?